save/record_instancereference.ts

```diff
@@ -121,21 +120,11 @@ export class RecordInstanceReference {
                     let iref = buf.readInt(maxOffset);
                     let stackedItemsNum = buf.readInt(maxOffset);
                     let changedEntriesNum = buf.readInt(maxOffset);
-                    let changedEntries: {
-                        propertiesNum: number;
-                        properties: {
-                            flag: number;
-                            // @TODO: More strict typing here?
-                            value: any;
-                        }[];
-                    }[] = [];
+                    let changedEntries: PropertyCollection[] = [];
                     for (let j = 0; j < changedEntriesNum; ++j) {
                         if (buf.offset - startOffset > record.dataSize) {/* console.log('Invalid object', record, this); */ return;}
                         let props = getProps(buf, startOffset + record.dataSize);
-                        changedEntries.push({
-                            propertiesNum: props[0],
-                            properties: props[1],
-                        });
+                        changedEntries.push(props);
```

Comment:
There's no need to store `props` locally with the change in method signature, so this could instead be directly pushed to the `changedEntries` array via `changedEntries.push(getProps(buf, startOffset + record.dataSize));`